scrapy_zyte_api/providers.py

```diff
@@ -116,6 +122,20 @@ async def __call__(  # noqa: C901
             options_name = f"{kw}Options"
             if item_type not in to_provide_stripped and options_name in zyte_api_meta:
                 del zyte_api_meta[options_name]
```

Comment:
When Zyte API's Auto-Configurator would be released, we won't have the need for `web_poet.HttpOrBrowserResponse` since we can receive the corresponding extraction source from the API Response.

More or less, the same code would be used after some slight tweaks.

Reply:
I think we'll still need HttpOrBrowserResponse dependency, as a way to trigger this API behavior.

Reply:
I was thinking that it would be triggered by the new Zyte API param that would return the extraction source that was used. For example, having `{..., "anyResponse": True}` in the Zyte API request would return `httpResponseBody`+`httpResponseHeaders` or `browserHtml` in Zyte API's response.

Reply:
Right, and this "anyResponse" parameter is going to be included in the API request sent by scrapy-zyte-api if a page object (or a callback) uses HttpOrBrowserResponse dependency, and won't be included otherwise, right?

Reply:
Aha, yes I see your point now 👍